CefSharp.BrowserSubprocess.Core/CefAppUnmanagedWrapper.cpp

```diff
@@ -405,17 +405,40 @@ namespace CefSharp
                 JavascriptAsyncMethodCallback^ callback;
                 if (rootObjectWrapper->TryGetAndRemoveMethodCallback(callbackId, callback))
                 {
-                    auto success = argList->GetBool(2);
-                    if (success)
-                    {
-                        callback->Success(DeserializeV8Object(argList, 3));
-                    }
-                    else
+
+                    auto frame = browser->GetFrame(frameId);
+                    if (frame.get())
                     {
-                        callback->Fail(argList->GetString(3));
+                        auto context = frame->GetV8Context();
+
+                        if (context.get() && context->Enter())
+                        {
+                            try
+                            {
+                                auto success = argList->GetBool(2);
+                                if (success)
+                                {
+                                    callback->Success(DeserializeV8Object(argList, 3));
+                                }
+                                else
+                                {
+                                    callback->Fail(argList->GetString(3));
+                                }
+                                //dispose
+                                delete callback;
```

Comment:
If it somehow fails this wont be called, you should place it inside the finally block